### PR TITLE
Android: Unblock Login Screen

### PR DESCRIPTION
**Pull request by edwinduong**
LoginScreen now disables fields and buttons instead of freezing on login or register request

### CODE REVIEW

File: android/app/src/main/java/io/github/parkwithease/parkeasy/ui/login/LoginViewModel.kt [hunk 1]
```diff
@@ -4,50 +4,65 @@ import androidx.lifecycle.ViewModel
...
 import io.github.parkwithease.parkeasy.model.RegistrationCredentials
 import io.github.parkwithease.parkeasy.model.ResetCredentials
 import javax.inject.Inject
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.asStateFlow
-import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.launch
```
Comment (TanoVip123) [inline android/app/src/main/java/io/github/parkwithease/parkeasy/ui/login/LoginViewModel.kt:18]: Why do we change  from launch to runBlocking?

**Comment by TanoVip123** [comment]
Could you please clarify what this change is for? When I login, it go to the user profile page as normal ...

File: android/app/src/main/java/io/github/parkwithease/parkeasy/ui/login/LoginViewModel.kt (hunk 1 above)
Comment (edwinduong) [reply android/app/src/main/java/io/github/parkwithease/parkeasy/ui/login/LoginViewModel.kt:18]: `runBlocking` blocks the UI layer, so the UI layer becomes unresponsive, by switching to `CoroutineScope(ioDispatcher).launch`, the UI is able to be updated and interacted with

**Comment by edwinduong** [comment]
> Could you please clarify what this change is for? When I login/sign up, it goes to the user profile page as normal, so not sure what this PR is doing

This PR is working on a suggestion from #111 where the UI becomes unresponsive while trying to login or register, it isn't noticeable when testing with a local dev environment since the server is fast enough to the point where we don't notice, but simulating latency with a `delay` will show the login fields and button being disabled

File: android/app/src/main/java/io/github/parkwithease/parkeasy/ui/login/LoginViewModel.kt [hunk 2]
```diff
@@ -4,50 +4,65 @@ import androidx.lifecycle.ViewModel
...
 
+    private val _formEnabled = MutableStateFlow(true)
+    val formEnabled = _formEnabled.asStateFlow()
+
     private val _message = MutableStateFlow(Event.initial(""))
     val message = _message.asStateFlow()
 
     fun onLoginPress(email: String, password: String) {
-        runBlocking {
+        CoroutineScope(ioDispatcher).launch {
```
[the remark this answers is not in the document]
Comment (edwinduong) [reply android/app/src/main/java/io/github/parkwithease/parkeasy/ui/login/LoginViewModel.kt:37]: Moved to #181

**Pull request merged**